hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlConnector.java

```diff
@@ -424,6 +393,38 @@ default List<String> getPrimaryKey(Table table) {
         throw new UnsupportedOperationException("PRIMARY KEY not supported by connector: " + typeName());
     }
 
+    interface DagBuildContext {
+        /**
+         * Returns the {@link DAG} that's being created.
+         */
+        DAG getDag();
+
+        /**
+         * Returns the context table. It is:<ul>
+         *     <li>for scans, it's the scanned table
+         *     <li>for DML, it's the target table
```

Comment:
in the implementation `table` is allowed to be null. Nullability is not explicitly defined in the interface

Reply:
the javadoc should define when `getTable()` can be null and what does it mean.